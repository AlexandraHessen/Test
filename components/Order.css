.Order{
    background-color: #e6f7dd;
    padding: 20px;
}

/* .OrderMade{
    background-color: #e6f7dd;
    padding: 20px;
    border-radius: 40px;
    border: solid 2px #8ccb67;
    text-align: center;
} */

.LabelOrder{
    /* font-weight: bold; */
    /* margin: 20px; */
}

.InputOrder{
    display: block;
    width: 30%;
    height: 40px;
    padding: 10px;
    font-size: 14px;
    font-family: 'Rubik', sans-serif;
    line-height: 1.428571429;
    color:  #363636;
    vertical-align: middle;
    background-color: #fff;
    background-image: none;
    border: 1px solid #fff;
    border-radius: 40px;
    /* margin:  0 0 20px 20px; */
    margin-bottom: 20px;
    font-weight: bold;
   
}

.InputOrder:focus, .TextareaOrder:focus{
    outline: 0;
}

input::placeholder {
    font-weight: normal;
    color: #969696;
  }

/* @-webkit-keyframes autofill {
    to {
      color: inherit;
      background: #fff;
    }
  }
  
  input:-internal-autofill-selected {
    -internal-animation-name: autofill;
    -internal-animation-fill-mode: both;
  } */

  input {outline: none;}
input:-webkit-autofill {
    -webkit-box-shadow: inset 0 0 0 50px #fff !important; 
} 

.TextareaOrder{
    display: block;
    height: auto;
    width: 50%;
    height: 40px;
    padding: 10px;
    font-size: 14px;
    font-family: 'Rubik', sans-serif;
    line-height: 1.428571429;
    color:  #363636;
    vertical-align: middle;
    background-color: #fff;
    background-image: none;
    border: 1px solid #fff;
    border-radius: 40px;
    margin-bottom: 20px;
    resize: none;
    font-weight: bold;
}

.CheckoutButton{
    border: solid 2px #8ccb68;
    font-family: 'Rubik', sans-serif;
    color:  #363636;
    background-color: #fff;
    text-transform: uppercase;
    font-size: 12px;
    border-radius: 40px;
    padding: 16px 25px;
    font-weight: bold;
    display: block;
    margin: auto;
    margin-bottom: 20px;
    cursor: pointer;
    /* outline: none !important; */
    /* border: solid 2px #8ccb68;
    padding: 8px 13px;
    border-radius: 40px;
    cursor: pointer;
    position: relative;
    background-color: #f9f9f9;
    font-family: 'Rubik', sans-serif;
    font-weight: bold;
    margin-bottom: 25px; */
    /* display: block;
    margin-left: auto;
    margin-right: 0; */
}

.Error{
    color:red
}

input[type="button"]:disabled{
    cursor: not-allowed;
    color: #c6c6c6;
    border: solid 2px #c6c6c6;
    /* background: currentColor; */
    background-color: #f9f9f9;
 }

 input:-internal-autofill-selected{
    background-color: -internal-light-dark(#fff, rgba(50, 185, 8, 0.4)) !important;
 }

 @media all and (max-width: 500px){
    .InputOrder{
        width: 100%;
    }

    .TextareaOrder{
        width: 100%;
    }

}
 