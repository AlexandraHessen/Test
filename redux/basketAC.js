// ---------------- ActionCreator ----------------//
const ADD_PRODUCT = "ADD_PRODUCT"
const DEL_PRODUCT = "DEL_PRODUCT"
const CLEAR_BASKET = "CLEAR_BASKET" //пользователь сам очищает корзину при желании

//название action type

//функция для dispatch (из файла CounterButton) которая возвращает хэш с action type 
// называется ActionCreator

const add_product = function(productId, objProductInfo) {
    return {
      type: ADD_PRODUCT,
      productId: productId,
      objProductInfo: objProductInfo
    };
  }
  
  const del_product = function(productId) {
    return {
      type: DEL_PRODUCT,
      productId: productId,
    };
  }

  const clear_basket = function() {
    return {
      type: CLEAR_BASKET
    };
  }

export {
    add_product, ADD_PRODUCT,
    del_product, DEL_PRODUCT,
    clear_basket, CLEAR_BASKET,
  }