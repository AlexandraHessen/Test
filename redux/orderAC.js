// ---------------- ActionCreator ----------------//
const ADD_ORDER = "ADD_ORDER"

//название action type

//функция для dispatch (из файла CounterButton) которая возвращает хэш с action type 
// называется ActionCreator

const add_order = function(objCustomerInfo, objOrderInfo) {
    return {
      type: ADD_ORDER,
      objCustomerInfo: objCustomerInfo,
      objOrderInfo: objOrderInfo
    };
  }

export {
    add_order, ADD_ORDER,
  }