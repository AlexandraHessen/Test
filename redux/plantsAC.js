// ---------------- ActionCreator ----------------//
const PLANTS_LOADING='PLANTS_LOADING';
const PLANTS_ERROR='PLANTS_ERROR';
const PLANTS_SET='PLANTS_SET';
//название action type

//функция для dispatch (из файла CounterButton) которая возвращает хэш с action type 
// называется ActionCreator
const plantsLoadingAC=function() {
  return {
    type: PLANTS_LOADING,
  };
}

const plantsErrorAC=function() {
  return {
    type: PLANTS_ERROR,
  };
}

const plantsSetAC=function(plants) {
  return {
    type: PLANTS_SET,
    plants:plants,
  };
}

export {
  plantsLoadingAC,PLANTS_LOADING,
  plantsErrorAC,PLANTS_ERROR,
  plantsSetAC,PLANTS_SET,
}
