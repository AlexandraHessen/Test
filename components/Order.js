"use strict";

import React from 'react';
import PropTypes from 'prop-types';

import './Order.css'

class Order extends React.PureComponent{
    state={
        nameError: 'Поле "Имя" обязательно для заполнения!',
        surnameError: 'Поле "Фамилия" обязательно для заполнения!',
        telError: "Введите корректный телефон!",
        emailError: "Введите корректный E-mail!",
    }

    render(){
        return(
            <div className="Order">
                <h1>Оформление заказа:</h1>
                <div>
                    <label htmlFor="name" className="LabelOrder">Имя</label>
                    <input type="text" name="name" className="InputOrder"></input>
                </div>
                <div>
                    <label htmlFor="surname" className="LabelOrder">Фамилия</label>
                    <input type="text" name="surname" className="InputOrder"></input>
                </div>
                <div>
                    <label htmlFor="tel" className="LabelOrder">Телефон</label>
                    <input type="text" name="tel" className="InputOrder"></input>
                </div>
                <div>
                    <label htmlFor="email" className="LabelOrder">Email</label>
                    <input type="text" name="email" className="InputOrder"></input>
                </div>
                <div>
                    <label htmlFor="delivery" className="LabelOrder">Способ доставки</label>
                    <select name="delivery" className="InputOrder">
                        <option value = {0}>Самовывоз</option>
                        <option value = {1}>Почтой</option>
                        <option value = {2}>Курьером по Минску</option>
                    </select>
                </div>
                <div>
                    <label htmlFor="payment" className="LabelOrder">Способ оплаты</label>
                    <select name="payment" className="InputOrder">
                        <option value = {0}>Наличными</option>
                        <option value = {1}>Пластиковой картой</option>
                        <option value = {2}>ЕРИП</option>
                    </select>
                </div>
                <div>
                    <label htmlFor="note" className="LabelOrder">Примечание</label>
                    <textarea name="note" className="TextareaOrder"></textarea>
                </div>
                <input type="button" value="Оформить заказ" className="CheckoutButton"></input>
            </div>
        )
    }
}
export default Order;
