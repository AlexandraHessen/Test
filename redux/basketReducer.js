// ---------------- САМА ФУНКЦИЯ REDUСER  ----------------//
// REDUCER не делает запрос в Ajax, логика должна быть в компоненте
import { ADD_PRODUCT, DEL_PRODUCT, CLEAR_BASKET } from './basketAC'; //название action type


const initState={ // начальный state
    productsInBasket: {}
    //КЛЮЧИ productId, objProductInfo
    // 1: {code: 1, name: 'Замиокулькас'}

//   status: 0, // 0 - ничего не началось, 1 - идёт загрузка, 2 - была ошибка, 3 - данные загружены
//   data: null, //даные которые мы загрузили по сети

}

function basketReducer(state=initState,action) {
  switch (action.type) {

    case ADD_PRODUCT: {
      console.log('action:',action);
      console.log('state до обработки редьюсером:',state);
      let newState={...state, //старый state
        productsInBasket:{...state.productsInBasket, //заменить старые продукты на теже +
          // ПРИЛЕТИТ какой продукт (productid) изменить и насколько(addvalue)
          [action.productId]:action.objProductInfo //+
          //[action.counterid] - В нужном продукте 
          // state.productsInBasket[action.productid] - его старое значение 
          // action.addvalue - изменить на прилетевшее
        }
      };
      console.log('state после обработки редьюсером:',newState);
      return newState;
    }

    case DEL_PRODUCT: {
      let newState = {...state};
      delete newState.productsInBasket[action.productId];
      return newState;
    }

    case CLEAR_BASKET:{
      let newState = {...state};
      newState.productsInBasket={};
      return newState;
    }

    // case PLANTS_ERROR: {
    //   let newState={
    //     status:2,
    //     data:null,
    //   };
    //   return newState;
    // }

    // case PLANTS_SET: {
    //   let newState={
    //     status:3,
    //     data:action.plants, //загруженные данные
    //   };
    //   return newState;
    // }
    
//обязатнльно пишем на тот случай если пришел action который не относится к этому reducery
    default:
      return state;
  }
}

export default basketReducer;
