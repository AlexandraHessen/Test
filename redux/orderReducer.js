// ---------------- САМА ФУНКЦИЯ REDUСER  ----------------//
// REDUCER не делает запрос в Ajax, логика должна быть в компоненте
import { ADD_ORDER } from './orderAC'; //название action type

const initState={ // начальный state
    orders: []
}

function orderReducer(state=initState,action) {
  switch (action.type) {
    case ADD_ORDER: {
      let newOrder=[...state.orders]
      newOrder.push( { objCustomerInfo: action.objCustomerInfo, products: action.objOrderInfo } );
      let newState={...state, orders: newOrder};

      return newState;
    }

              //обязатнльно пишем на тот случай если пришел action который не относится к этому reducery
    default:
      return state;
  }
}

export default orderReducer;