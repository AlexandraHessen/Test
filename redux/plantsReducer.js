// ---------------- САМА ФУНКЦИЯ REDUСER  ----------------//
// REDUCER не делает запрос в Ajax, логика должна быть в компоненте
import { PLANTS_LOADING, PLANTS_ERROR, PLANTS_SET } from './plantsAC'; //название action type


const initState={ // начальный state

  status: 0, // 0 - ничего не началось, 1 - идёт загрузка, 2 - была ошибка, 3 - данные загружены
  data: null, //даные которые мы загрузили по сети

}

function plantsReducer(state=initState,action) {
  switch (action.type) {

    case PLANTS_LOADING: {
      let newState={
        status:1,
        data:null,
      };
      return newState;
    }

    case PLANTS_ERROR: {
      let newState={
        status:2,
        data:null,
      };
      return newState;
    }

    case PLANTS_SET: {
      let newState={
        status:3,
        data:action.plants, //загруженные данные
      };
      return newState;
    }
    
//обязатнльно пишем на тот случай если пришел action который не относится к этому reducery
    default:
      return state;
  }
}

export default plantsReducer;
