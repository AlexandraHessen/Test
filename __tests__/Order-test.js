"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import { createStore } from 'redux';
import combinedReducer from '../redux/reducers.js';
import { Provider } from 'react-redux';

import Order from '../components/Order';
                    //импортируем файл который будем тестировать

let store=createStore(combinedReducer);

test('Order', () => {
                            // создаём тестовую версию компонента который будем тестировать
                            const component = renderer.create(
    <Provider store={store}>
        <BrowserRouter>
            <Order/>
        </BrowserRouter>
    </Provider>
    );

                            // получаем снэпшот (HTML-снимок) компонента для сверки, чтобы вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

    
});
