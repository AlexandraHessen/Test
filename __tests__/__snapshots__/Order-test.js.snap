// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order 1`] = `
<div
  className="Order"
>
  <h1>
    Оформление заказа:
  </h1>
  <div>
    <label
      className="LabelOrder"
      htmlFor="name"
    >
      Имя
    </label>
    <input
      autoFocus={true}
      className="InputOrder"
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </div>
  <div>
    <label
      className="LabelOrder"
      htmlFor="surname"
    >
      Фамилия
    </label>
    <input
      className="InputOrder"
      name="surname"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </div>
  <div>
    <label
      className="LabelOrder"
      htmlFor="tel"
    >
      Телефон
    </label>
    <input
      className="InputOrder"
      name="tel"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="+37529111-22-33"
      type="text"
    />
  </div>
  <div>
    <label
      className="LabelOrder"
      htmlFor="email"
    >
      Email
    </label>
    <input
      className="InputOrder"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="email@gmail.com"
      type="text"
    />
  </div>
  <div>
    <label
      className="LabelOrder"
      htmlFor="delivery"
    >
      Способ доставки
    </label>
    <select
      className="InputOrder"
      name="delivery"
      onChange={[Function]}
    >
      <option
        value="pickup"
      >
        Самовывоз
      </option>
      <option
        value="post"
      >
        Почтой
      </option>
      <option
        value="courier"
      >
        Курьером по Минску
      </option>
    </select>
  </div>
  <div>
    <label
      className="LabelOrder"
      htmlFor="payment"
    >
      Способ оплаты
    </label>
    <select
      className="InputOrder"
      name="payment"
      onChange={[Function]}
    >
      <option
        value="cash"
      >
        Наличными
      </option>
      <option
        value="card"
      >
        Пластиковой картой
      </option>
      <option
        value="erip"
      >
        ЕРИП
      </option>
    </select>
  </div>
  <div>
    <label
      className="LabelOrder"
      htmlFor="note"
    >
      Примечание
    </label>
    <textarea
      className="TextareaOrder"
      name="note"
      onChange={[Function]}
    />
  </div>
  <input
    className="CheckoutButton"
    disabled={false}
    onClick={[Function]}
    type="button"
    value="Оформить заказ"
  />
</div>
`;
